gateway/api/localauth/register.go

```diff
@@ -0,0 +1,142 @@
+package localauthapi
+
+import (
+	"fmt"
+	"libhoop/log"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+	"github.com/hoophq/hoop/gateway/appconfig"
+	"github.com/hoophq/hoop/gateway/pgrest"
+	pglocalauthsession "github.com/hoophq/hoop/gateway/pgrest/localauthsession"
+	pgorgs "github.com/hoophq/hoop/gateway/pgrest/orgs"
+	pgusers "github.com/hoophq/hoop/gateway/pgrest/users"
+	"github.com/hoophq/hoop/gateway/storagev2/types"
+	"golang.org/x/crypto/bcrypt"
+)
+
+// If the system is set as single tenant,
+// we default the new user to the default organization.
+// Otherwise, a new organization is created for the user.
+func manageOrgCreation(user pgrest.User) (string, error) {
+	var tenancy string
+	if pgusers.IsOrgMultiTenant() {
+		tenancy = "multi-tenant"
+	} else {
+		tenancy = "single-tenant"
+	}
+	switch tenancy {
+	case "multi-tenant":
+		log.Debug("Creating new organization")
+		newOrgID, err := pgorgs.New().CreateOrGetOrg(fmt.Sprintf("%q Orgnization", user.Email), nil)
+		if err != nil {
+			log.Debugf("failed creating organization, err=%v", err)
+			return "", fmt.Errorf("Failed to create organization")
+		}
+		return newOrgID, nil
+	default:
+		// fetch default organization
+		org, totalUsers, err := pgorgs.New().FetchOrgByName("default")
+		if err != nil {
+			return "", fmt.Errorf("Failed to fetch default organization")
+		}
+		// if there is one user already, do not allow new users to be created
+		// it avoids a security issue of anyone being able to add themselves to
+		// the default organization. Instead, they should get an invitation
+		if totalUsers > 0 {
+			return "", fmt.Errorf("You can not access this instance. Please contact your administrator")
+		}
+
+		return org.ID, nil
+	}
+}
+
+func Register(c *gin.Context) {
```

Comment:
We need to be more defensive here and checking if the secret key is set (!= "") before allowing this route to be used.

Reply:
Agreed! I added it to the app configuration. When AUTH_METHOD is set to local, we should reinforce for a JWT secret.